numpy/core/function_base.py

```diff
@@ -349,40 +351,46 @@ def geomspace(start, stop, num=50, endpoint=True, dtype=None):
     >>> plt.show()
 
     """
-    if start == 0 or stop == 0:
+    start = asanyarray(start)
+    stop = asanyarray(stop)
+    if _nx.any(start == 0) or _nx.any(stop == 0):
         raise ValueError('Geometric sequence cannot include zero')
 
-    dt = result_type(start, stop, float(num))
+    dt = result_type(start, stop, float(num), _nx.zeros((), dtype))
```

Comment:
Does this behave differently than just passing `dtype` in directly?

Reply:
`dtype` can be `None` - I actually tried thinking of a quicker way.